Paper-API-Patches/0024-Add-stopsound-command.patch

```diff
@@ -0,0 +1,1291 @@
++    MUSIC_DRAGON("minecraft:music.dragon", MUSIC),
++    MUSIC_END("minecraft:music.end", MUSIC),
++    MUSIC_GAME("minecraft:music.game", MUSIC),
++    MUSIC_MENU("minecraft:music.menu", MUSIC),
++    MUSIC_NETHER("minecraft:music.nether", MUSIC),
++    RECORD_11("minecraft:record.11", RECORD),
++    RECORD_13("minecraft:record.13", RECORD),
++    RECORD_BLOCKS("minecraft:record.blocks", RECORD),
++    RECORD_CAT("minecraft:record.cat", RECORD),
++    RECORD_CHIRP("minecraft:record.chirp", RECORD),
++    RECORD_FAR("minecraft:record.far", RECORD),
++    RECORD_MALL("minecraft:record.mall", RECORD),
++    RECORD_MELLOHI("minecraft:record.mellohi", RECORD),
++    RECORD_STAL("minecraft:record.stal", RECORD),
++    RECORD_STRAD("minecraft:record.strad", RECORD),
++    RECORD_WAIT("minecraft:record.wait", RECORD),
++    RECORD_WARD("minecraft:record.ward", RECORD),
++    UI_BUTTON_CLICK("minecraft:ui.button.click"),
++    WEATHER_RAIN("minecraft:weather.rain", WEATHER),
++    WEATHER_RAIN_ABOVE("minecraft:weather.rain.above", WEATHER);
++
++    private static Sound[] SOUNDS;
++
++    private String id;
++    private Category category;
++
++    Sound(String id) {
++        this(id, Category.MASTER);
++    }
++
++    Sound(String id, Category category) {
++        this.id = id;
++        this.category = category;
++    }
++
++    public String getId() {
++        return id;
++    }
++
++    public Category getCategory() {
++        return category;
++    }
++
++    public static Sound fromId(String id) {
++        if (SOUNDS == null) {
++            SOUNDS = values();
++        }
++        if (!id.startsWith("minecraft:")) {
++            id = "minecraft:" + id;
++        }
++        for (Sound s : SOUNDS) {
++            if (s.getId().equalsIgnoreCase(id)) {
++                return s;
++            }
++        }
++        return null;
++    }
++
++    public enum Category {
++        MASTER,
++        MUSIC,
++        RECORD,
++        WEATHER,
++        BLOCK,
++        HOSTILE,
++        NEUTRAL,
++        PLAYER,
++        AMBIENT,
++        VOICE;
++
++        public static Category getCategory(String name) {
++            for (Category category : values()) {
++                if (category.name().equalsIgnoreCase(name)) {
++                    return category;
++                }
++            }
++            return Category.MASTER;
++        }
++
++        public static Sound.Category fromInt(int i) {
++            return values()[i];
++        }
++    }
+ }
+diff --git a/src/main/java/org/bukkit/command/CommandTarget.java b/src/main/java/org/bukkit/command/CommandTarget.java
+index 74f8fc6..0a33130 100644
+--- a/src/main/java/org/bukkit/command/CommandTarget.java
++++ b/src/main/java/org/bukkit/command/CommandTarget.java
+@@ -261,7 +261,7 @@ public class CommandTarget {
+     /**
+      * Represents the value of a selector argument
+      */
+-    class SelectorValue {
++    public static class SelectorValue {
```

Comment:
@mastercoms 
Quick fix for this: https://github.com/GlowstoneMC/Glowkit/pull/8/files#diff-955e9dd0ca1ee82e7a106f40fa04d1a5R1220